tests/lib/linter/Linter.test.js

```diff
@@ -0,0 +1,184 @@
+const { Linter } = require("../../../lib/linter");
+const { Rules } = require("../../../lib/rules");
+
+const config = {
+    rules: {
+        indentation: "error",
+        no_repetitive_step_keyword: "error",
+    },
+};
+
+const mockProblem = [
+    {
+        ruleId: "indentation",
+        type: "error",
+        message: "Indentation error",
+        location: {},
+    },
+];
+
+describe("class: Linter", () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+        jest.restoreAllMocks();
+    });
+    describe("method: lint", () => {
+        describe("without fix option", () => {
+            it("should return an empty array if no lint issues", () => {
+                const spyRunRules = jest
+                    .spyOn(Linter.prototype, "runRules")
+                    .mockReturnValue([]);
+
+                const linter = new Linter(config);
+                const result = linter.lint("");
```

Comment:
```suggestion
                const linter = new Linter(config);
                const result = linter.lint("");
```
Frequent use of initiating `Linter`. Repetitive code lines, can't we optimize them?

Reply:
we could move it to hooks but different config requirement makes it complicated. There will be multiple hooks and might make grouping tests not easy

Reply:
But I agree, tests need some trimming